modules/server/src/main/scala/higherkindness/compendium/parser/ProtocolParserService.scala

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2018-2019 47 Degrees, LLC. <http://www.47deg.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package higherkindness.compendium.parser
+
+import cats.effect.Sync
+import higherkindness.compendium.models.{MetaProtocol, Target}
+import higherkindness.compendium.models.parserModels.ParserResult
+
+trait ProtocolParserService[F[_]] {
+
+  def parse(protocol: Option[MetaProtocol], target: Target): F[ParserResult]
```

Comment:
I think we should keep this service simple and don't use `Option` arguments, especially if this could/must be handled at `CompendiumService` level, raising an error when `recoverProtocol` fails. What do you think?

Reply:
Yeah sure, but can it be done in the next PR? I would like to merge this code to avoid more infinite conflicts. Once the integration is merged, I can perform minor changes with fewer conflicts

Reply:
Ok!

Reply:
At the end I added this change in this PR